File: leetcode/228.py
"""
    File: 228.py
    Title: Summary Ranges
    Difficulty: Easy
"""

from typing import List


class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []
        if len(nums) == 1:
            return [str(nums[0])]

        ans = []
        a = 0
        for i in range(1, len(nums)):
            if nums[i] - nums[i - 1] == 1:
                pass
            else:
                if a == i - 1:
                    ans.append(str(nums[a]))
                else:
                    ans.append(f"{nums[a]}->{nums[i - 1]}")
                a = i
        if a == len(nums) - 1:
            ans.append(str(nums[a]))
        else:
            ans.append(f"{nums[a]}->{nums[len(nums) - 1]}")
        return ans

File: leetcode/228.go
package leetcode

import (
	"fmt"
	"strconv"
)

func summaryRanges(nums []int) []string {
	if len(nums) == 0 {
		return []string{}
	}

	ans := []string{}
	start := 0
	end := len(nums) - 1
	for i := 1; i <= end; i++ {
		if nums[i] == nums[i-1]+1 {
			continue
		} else {
			if i-start == 1 {
				ans = append(ans, strconv.Itoa(nums[start]))
			} else {
				ans = append(ans, fmt.Sprintf("%d->%d", nums[start], nums[i-1]))
			}
			start = i
		}
	}
	if start == end {
		ans = append(ans, strconv.Itoa(nums[start]))
	} else {
		ans = append(ans, fmt.Sprintf("%d->%d", nums[start], nums[end]))
	}
	return ans
}
